components/upload/upload-btn.component.ts

```diff
@@ -134,24 +134,26 @@ export class NzUploadBtnComponent implements OnInit, OnDestroy {
     if (this.options.filters) {
       this.options.filters.forEach(f => {
         filters$ = filters$.pipe(
-          switchMap(list => {
+          concatMap(list => {
             const fnRes = f.fn(list);
             return fnRes instanceof Observable ? fnRes : of(fnRes);
           })
         );
       });
     }
-    filters$.subscribe(
-      list => {
+    filters$
+      .pipe(
+        catchError(e => {
+          warn(`Unhandled upload filter error`, e);
+          return EMPTY;
+        })
+      )
+      .subscribe(list => {
         list.forEach((file: NzUploadFile) => {
           this.attachUid(file);
           this.upload(file, list);
         });
-      },
-      e => {
-        warn(`Unhandled upload filter error`, e);
-      }
-    );
+      });
```

Comment:
I think we don't have to use catchError operator :)

```ts
.subscribe({
  next: list => {...},
  error: e => {...}
});
```

Reply:
that's true, it just one of my way to do. but it's ok let refactor